servicetalk-concurrent-internal/src/main/java/io/servicetalk/concurrent/internal/DelayedCancellable.java

```diff
@@ -52,4 +52,12 @@ public void cancel() {
             oldCancellable.cancel();
         }
     }
+
+    /**
+     * Dereference the internal {@link Cancellable} and prevent future calls to
+     * {@link #delayedCancellable(Cancellable)}.
+     */
+    protected final void disableCancellable() {
```

Comment:
Is this useful anywhere else but for the concat case? 

Can the switch between `cancellable` and `subscription` be local to `DelayedCancellableThenSubscription`?

Reply:
`DelayedCancellableThenSubscription` is the only consumer of this API.

Options to accomplish dereference:
- `DelayedCancellable#cancel()` - this may involve additional offloading and complicates the control flow a bit.
- `DelayedCancellable#disableCancellable` - adds extra protected method to the API. The API
- `DelayedCancellableThenSubscription` to duplicate `DelayedCancellable` logic - logic isn't overly complicated so may not be a deal breaker, but more code duplication.

I'll update to the last option to avoid API change for review.